src/main/java/io/vertx/core/Vertx.java

```diff
@@ -345,6 +368,28 @@ static void clusteredVertx(VertxOptions options, Handler<AsyncResult<Vertx>> res
   void deployVerticle(Verticle verticle, DeploymentOptions options, Handler<AsyncResult<String>> completionHandler);
 
   /**
+   * Like {@link #deployVerticle(Verticle, DeploymentOptions, Handler)} but {@link Verticle} instance is created by
+   * invoking the default constructor of {@code verticleClass}.
+   */
+  @GenIgnore
+  void deployVerticle(Class<? extends Verticle> verticleClass, DeploymentOptions options, Handler<AsyncResult<String>> completionHandler);
+
+  /**
+   * Like {@link #deployVerticle(Verticle, DeploymentOptions, Handler)} but {@link Verticle} instance is created by
+   * invoking the {@code verticleSupplier}.
+   * <p>
+   * The supplier will be invoked as many times as {@link DeploymentOptions#getInstances()}.
+   * It must not return:
+   * <ul>
+   * <li>null</li>
+   * <li>the same instance twice</li>
+   * <li>instances of different verticle classes</li>
+   * </ul>
+   */
+  @GenIgnore
+  void deployVerticle(Supplier<? extends Verticle> verticleSupplier, DeploymentOptions options, Handler<AsyncResult<String>> completionHandler);
```

Comment:
do we really need the `<? extends Verticle>` most users will use a lambda here (I'm not a fan of wildcard generics).